src/functions-and-arrays.js

```diff
@@ -1,38 +1,83 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers (num1, num2) {
+  if (num1>num2) {
+    return num1;
+  } else {
+    return num2;
+    }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) {
+  if (words.length === 0) {
+    return null;
+  }
+  let word = '';
+  for (let i=0; i<words.length; i++ ) {
+    if (word.length<words[i].length){
+      word = words[i];
+    }
+  }
+  return word;
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+
+function sumNumbers(numbers) {
+  if (numbers.length === 0 ) {
+    return zero;
+  }
+  let sum = 0;
+  for (let i=0; i<numbers.length; i++){
+    sum = sum +numbers[i];
+  }
+  return sum;
+}
+
+
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(numbers) {}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbersAvg) {
+  if (numbersAvg.length === 0 ) {
+    return null;
+  }
+  let sum = sumNumbers(numbersAvg)
+    return sum / numbersAvg.length;
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(something) {
+  if (averageWordLength.length === 0){
+    return null;
+  }
+    arrLength = [];
+    for (let i = 0; i < something.length; i++) {
+      arrLength.push(something[i].length);
+    }
```

Comment:
This works and what's nice about this is that you are using the other function. Although I would say that in here it's actually not so necessary because it needs to create a whole new array and that's not so efficient.  Using reduce would be best, but you can also just do something like Jan did:
``` 
let sum = 0;
  for (let i = 0; i < words.length; i++) {
    sum += words[i].length;
  }
  return sum / words.length;
```